### PR TITLE
API Error Handling. Solves #399

### PR DESCRIPTION
**Pull request by caleeli**
API Error Handling.

image

image

### CODE REVIEW

File: resources/assets/sass/app.scss [hunk 1]
```diff
@@ -411,5 +411,5 @@ table.vuetable {
   position: fixed;  
   left:30%;
   right:30%;
-
+  z-index: 1051;
```
Comment (danloa) [inline resources/assets/sass/app.scss]: wouldn't it better to have a more meaningful number like 9999 and set it as the maximum to be used for the z-index of all our system?

File: ProcessMaker/Http/Controllers/Api/Workflow/ActivityController.php [hunk 2]
```diff
@@ -14,7 +14,7 @@ class ActivityController extends Controller
     public function complete(Request $request, Process $process, Instance $instance, Delegation $token)
     {
         if ($token->thread_status === Delegation::THREAD_STATUS_CLOSED) {
-            return abort(404);
+            return abort(404, 'Task already closed');
```
Comment (velkymx) [inline ProcessMaker/Http/Controllers/Api/Workflow/ActivityController.php]: Let's use 410 Gone
"Indicates that the resource requested is no longer available and will not be available again."

File: resources/assets/js/management/groups/index.js [hunk 3]
```diff
@@ -35,8 +35,6 @@ new Vue({
 
       })
       .catch((err) => {
```
Comment (velkymx) [inline resources/assets/js/management/groups/index.js]: Do we need the catch at all?

File: resources/assets/sass/app.scss (hunk 1 above)
Comment (caleeli) [reply resources/assets/sass/app.scss]: Change to sass variable: $alertZIndex

File: ProcessMaker/Http/Controllers/Api/Workflow/ActivityController.php (hunk 2 above)
Comment (caleeli) [reply ProcessMaker/Http/Controllers/Api/Workflow/ActivityController.php]: Fixed

**Comment by caleeli** [comment]
The observations related to the error code, catches and z-index where fixed.

File: resources/assets/js/management/roles/index.js [hunk 4]
```diff
@@ -36,12 +36,7 @@ new Vue({
...
-
-      })
-      .catch((err) => {
-        // @todo Replace with new flashy errors?
-        alert('There was a problem creating the role.')
-      })
+      });
     }
   }
 })
```
Comment (danloa) [inline resources/assets/js/management/roles/index.js]: add a line at the end of the file

Comment (caleeli) [reply resources/assets/js/management/roles/index.js]: fixed

**Pull request merged**